### PR TITLE
[0.11.0] Refactor: List-detail Hook

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
## Code Changes
- `useBreakpoint` hook
  - `Breakpoint` enum
- `useListDetail` hook
  - Classes
  - Learn/Teach/Manage Electives
  - Search Students/Teachers/Documents

## Fixes
- The header part Document Details Card has the wrong color

## Preview
Preview available at https://pr1.mysk.school/.

### CODE REVIEW

**Review by smartwhatt — APPROVED** [review]
LGTM!!

**Review by pixelpxed — APPROVED** [review]
I see that you've simplified a lot of code to be more readable, LGTM!

File: utils/helpers/search/useListDetail.ts
Comment (Sun-52) [inline utils/helpers/search/useListDetail.ts:1]: Making a hook is very good for reducing repetitive code and logic and making the code easier to make changes to in the future.

**Review by Sun-52 — APPROVED** [review]
LGTM

**Pull request merged**